google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -381,12 +381,39 @@ async def _perform_refresh(self) -> asyncio.Task:
 
         logger.debug("Entered _perform_refresh")
 
-        self._current = self._loop.create_task(self._get_instance_data())
-        # Ephemeral certificate expires in 1 hour, so we schedule a refresh to happen in 55 minutes.
+        refresh_task = self._loop.create_task(self._get_instance_data())
 
-        self._next = self._loop.create_task(self._schedule_refresh())
+        def _refresh_callback(task: asyncio.Task) -> None:
+            try:
+                task.result()
+            except Exception as e:
+                logger.warn(
+                    "An error occurred while performing refresh. Retrying immediately.",
+                    e,
+                )
+                instance_data = None
+                try:
+                    instance_data = self._current.result()
+                except Exception:
+                    # Current result is invalid, no-op
+                    logger.debug("Current instance data is invalid.")
+                if (
+                    instance_data is None
+                    or instance_data.expiration < datetime.datetime.now()
+                ):
+                    self._current = refresh_task
```

Comment:
Should this be `task ` or `refresh_task`?

Reply:
either technically works since it's a closure, but task is probably better